unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -36,26 +34,65 @@ public class FriendsController : IFriendsController
         public event Action<FriendRequest> OnFriendRequestReceived;
         public event Action<FriendRequest> OnSentFriendRequestApproved;
 
-        public static void CreateSharedInstance(IFriendsApiBridge apiBridge)
-        {
-            i = new FriendsController(apiBridge);
-        }
+        private CancellationTokenSource controllerCancellationTokenSource;
 
-        public FriendsController(IFriendsApiBridge apiBridge)
+        public FriendsController(IFriendsApiBridge apiBridge, IRPCSocialApiBridge rpcSocialApiBridge)
         {
             this.apiBridge = apiBridge;
-            apiBridge.OnInitialized += Initialize;
             apiBridge.OnFriendNotFound += FriendNotFound;
             apiBridge.OnFriendWithDirectMessagesAdded += AddFriendsWithDirectMessages;
             apiBridge.OnUserPresenceUpdated += UpdateUserPresence;
             apiBridge.OnFriendshipStatusUpdated += HandleUpdateFriendshipStatus;
             apiBridge.OnTotalFriendRequestCountUpdated += UpdateTotalFriendRequests;
             apiBridge.OnTotalFriendCountUpdated += UpdateTotalFriends;
-            apiBridge.OnFriendRequestsAdded += AddFriendRequests; // TODO (NEW FRIEND REQUESTS): remove when we don't need to keep the retro-compatibility with the old version
-            apiBridge.OnFriendRequestReceived += ReceiveFriendRequest;
+
+            this.socialApiBridge = rpcSocialApiBridge;
+
+            socialApiBridge.OnFriendAdded += AddFriend;
+            socialApiBridge.OnFriendRemoved += InternalRemoveFriend;
+            socialApiBridge.OnFriendRequestAdded += AddFriendRequest;
+            socialApiBridge.OnFriendRequestRemoved += RemoveFriendRequest;
         }
 
-        private void Initialize(FriendshipInitializationMessage msg)
+        public void Initialize()
+        {
+            controllerCancellationTokenSource = new CancellationTokenSource();
+
+            socialApiBridge.InitializeClient(controllerCancellationTokenSource.Token).Forget();
+            socialApiBridge.InitializeFriendshipsInformation(controllerCancellationTokenSource.Token).ContinueWith(this.InitializeFriendships).Forget();
+            socialApiBridge.InitializeSubscriptions(controllerCancellationTokenSource.Token).Forget();
+        }
+
+        public void Dispose()
+        {
+            controllerCancellationTokenSource.Cancel();
+            controllerCancellationTokenSource.Dispose();
+        }
+
+        private void RemoveFriendRequest(string userId)
+        {
+            this.friendRequests.Remove(userId);
+        }
+
+        // TODO (Joni): Replace by AddFriendRequests, this is just for successful compilation
+        private void AddFriendRequest(FriendRequest friendRequest)
+        {
+            this.friendRequests.Add(friendRequest.FriendRequestId, friendRequest);
```

Comment:
you may introduce a KeyExistsException.. you can replace it by doing: `friendRequests[id] = friend;`